module/zfs/dmu_send.c

```diff
@@ -1537,7 +1541,8 @@ restore_object(struct restorearg *ra, objset_t *os, struct drr_object *drro)
 		}
 	}
 
-	tx = dmu_tx_create(os);
+	count = DN_BONUSLEN_TO_COUNT(drro->drr_bonuslen);
+	tx = dmu_tx_create_dn_count(os, count);
```

Comment:
I've explained poorly.  Right, it doesn't matter as long we've safely stashed it away with dmu_tx_hold_() for consistent use latter.  So I think we're in completely agreement.  I was just saying the initial prototype patch ran in to problems because it naively pulled the size from the property as needed.  That ended up causing some ASSERTs when it changed at the wrong time.  I'm sure @nedbass could have done a better job explaining.  :)